modules/social_features/social_event/modules/social_event_an_enroll/src/Controller/EventAnEnrollController.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+namespace Drupal\social_event_an_enroll\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Url;
+use Drupal\node\NodeInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\social_event_managers\SocialEventManagersAccessHelper;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+
+/**
+ * Class EventAnEnrollController.
+ *
+ * @package Drupal\social_event_an_enroll\Controller
+ */
+class EventAnEnrollController extends ControllerBase {
+
+  /**
+   * The route match.
+   *
+   * @var \Drupal\Core\Routing\RouteMatchInterface
+   */
+  protected $routeMatch;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * SocialTopicController constructor.
+   *
+   * @param \Drupal\Core\Routing\RouteMatchInterface $routeMatch
+   *   The route match object.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity type manager.
+   */
+  public function __construct(RouteMatchInterface $routeMatch, EntityTypeManagerInterface $entityTypeManager) {
+    $this->routeMatch = $routeMatch;
+    $this->entityTypeManager = $entityTypeManager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('current_route_match'),
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * Determines if user has access to enroll form.
+   */
+  public function enrollAccess(NodeInterface $node) {
+    $config = $this->config('social_event_an_enroll.settings');
+    $is_global_enabled = $config->get('event_an_enroll');
+    $is_event = $node->getType() === 'event';
+    $is_public = $node->get('field_content_visibility')->getString() === 'public';
+    $is_event_an_enroll = !empty($node->get('field_event_an_enroll')->value);
+    if ($is_global_enabled && $is_event && $is_public && $is_event_an_enroll) {
```

Comment:
You are right, but the difference between those two are that in `social_event_an_enroll_is_enabled()` we use global `\Drupal::config` and in `enrollAccess` we use dependency injection.
And it's hard to use only `enrollAccess` since `EventAnEnrollController::__construct` needs `$routeMatch` and `$entityTypeManager`. 
If yo think it's ok to call `social_event_an_enroll_is_enabled()` in `enrollAccess` I can fix that.

Reply:
I think the benefits of simply calling the function `social_event_an_enroll_is_enabled` outweigh the negatives of calling a static function from a class context.

We don't yet do any testing where not having this dependency injection is a problem and it ensures that if we ever change the requirements for `social_event_an_enroll_is_enabled` then this controller will also work as expected. 

If we really need  dependency injection at some point then we can always think about moving things like `social_event_an_enroll_is_enabled` into a service so that it can be used in both places.